src/component/marker/MarkerComponent.ts

```diff
@@ -107,12 +108,33 @@ export class MarkerComponent extends Component {
                 return this.renderHash(args);
             })
             .subscribe(this._container.glRenderer.render$);
+
+        this._disposableMapillaryObject = null;
+        this.configuration$.subscribe((conf: IMarkerConfiguration) => {
```

Comment:
The external state (`_disposableMapillaryObject`) can be avoided by doing (pseudo):

`this._configurationSubscription = this.configuration$.
    filter(conf.mapillaryObject)
    flatMapLatest<Marker>(return this._navigator.vectorTilesService.mapillaryObjects$.map<Marker>(create marker))
    subscribe(do things with marker)`